stats/views.py

```diff
@@ -78,6 +78,13 @@ def app_stats_api(request):
     grad_year_count_confirmed = Application.objects.filter(status=APP_CONFIRMED).values('graduation_year') \
         .annotate(applications=Count('graduation_year'))
 
+    degree_count = Application.objects.all().values('degree') \
+        .annotate(applications=Count('degree')) \
+        .order_by('-applications')[:10]
+    degree_count_confirmed = Application.objects.filter(status=APP_CONFIRMED).values('degree') \
+        .annotate(applications=Count('degree')) \
+        .order_by('-applications')[:10]
+
```

Comment:
This is starting to have too many queries that look like the same. Any chance we can factor all those into a single query? I'm afraid on performance here and pushing too hard on the DB.

Reply:
[Query sets are lazy](https://docs.djangoproject.com/en/3.0/topics/db/queries/#querysets-are-lazy), ones you call them they call the DB. If we put it in one "query" it really doesn't matter, django will call them one by one.

Reply:
Yes, but you have several query sets here, which translates to several DB queries on the background. Making one query that returns degrees/university/food... and then processing the dictionary to generate the aggregations may actually perform better as it runs in a single query and reduces the amount of times it requires to connect to the DB.